packages/test/test-version-utils/src/compatConfig.ts

```diff
@@ -154,14 +154,23 @@ const genFullBackCompatConfig = (): CompatConfig[] => {
 	// Extract the major version of the package published, in this case it's the x in 2.0.0-internal.x.y.z.
 	// This first if statement turns the internal version to x.y.z
 	let semverInternal: string | undefined;
-	if (pkgVersion.startsWith("2.0.0-internal.")) {
-		semverInternal = pkgVersion.split("internal.")[1];
-	} else if (pkgVersion.startsWith("2.0.0-dev.")) {
-		semverInternal = pkgVersion.split("dev.")[1];
+	let version: string = pkgVersion;
+	// This grabs the code version to find the backwards compatible options.
+	const codeVersion = process.env.SETVERSION_CODEVERSION;
+	if (
+		codeVersion !== undefined &&
+		(codeVersion.startsWith("2.0.0-internal.") || codeVersion.startsWith("2.0.0-dev."))
+	) {
+		version = codeVersion;
+	}
+	if (version.startsWith("2.0.0-internal.")) {
+		semverInternal = version.split("internal.")[1];
+	} else if (version.startsWith("2.0.0-dev.")) {
+		semverInternal = version.split("dev.")[1];
```

Comment:
I worry about this code. Can we use the version-tools package for some of this? It has functions to decode/encode internal versions, validate them, etc, plus tests. I think with those APIs, this code looks more like this:

```ts
const [publicVersion, semverInternal, prereleaseIdentifier] = fromInternalScheme(codeVersion, true, true);
```

Reply:
Relevant source: https://github.com/microsoft/FluidFramework/blob/547789bec19e64a7df2a04bb251c7c8939f6db8a/build-tools/packages/version-tools/src/internalVersionScheme.ts#L56